### PR TITLE
For #6531 new verifyTabsTrayListTest UI Smoke test and other refactoring work

### PR DESCRIPTION
**Pull request by AndiAJ**
• Restore the verifyTabsOrder function and minor refactoring work
• New `verifyTabsTrayListTest ` UI Smoke test ✅ Successfully passed 40x on Firebase 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/TabsTrayRobot.kt [hunk 1]
```diff
@@ -16,17 +16,31 @@ import org.hamcrest.Matchers.allOf
...
+                            ),
+                            withParentIndex(tab)
+                        )
+                    )
+                )
+            )
+        }
+    }
+
+    fun verifyCloseTabsButton(numberOfOpenedTabs: Int) {
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/focus/activity/robots/TabsTrayRobot.kt:36]: nit: verifyCloseTabButton

Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/TabsTrayRobot.kt:36]: Also, I would create a simple matcher for the close tab button, in relation to a given tab (given by a parameter) which would be reusable. And for the testVisitingMultipleSites test, I would separate the close tab button (and other tabs tray elements) validation in a new test (to match testRail).

Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/TabsTrayRobot.kt]: Done!

**Comment by sv-ohorvath** [comment]
Looks better to me, thanks!

**Review by AaronMT — APPROVED** [review]

**Pull request merged**